gxa/src/test/java/uk/ac/ebi/atlas/resource/ContrastImageControllerWIT.java

```diff
@@ -41,54 +50,72 @@
     private static final String GSEA_PLOT_IMAGE_URL_TEMPLATE =
             "/external-resources/".concat(RNASEQ_EXPERIMENT_ACCESSION).concat("/{0}/gsea_{1}.png");
 
+    @Inject
+    private DataSource dataSource;
+
     @Autowired
     private WebApplicationContext wac;
+
     private MockMvc mockMvc;
 
+    @BeforeAll
+    void populateDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-fixture.sql"));
+        populator.execute(dataSource);
+    }
+
+    @AfterAll
+    void cleanDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-delete.sql"));
+        populator.execute(dataSource);
+    }
+
     @BeforeEach
     void setUp() {
         this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
     }
 
     @Test
-    void responseForGseaPlotsShouldBeNonEmpty() throws Exception {
+    void responseForGseaPlotsShouldNotBeEmpty() throws Exception {
         responseAssertions(
-                this.mockMvc.perform(get(MessageFormat.format(GSEA_PLOT_IMAGE_URL_TEMPLATE, "g6_g2", "go"))));
```

Comment:
Interesting that you chose to stick to specific experiment IDs here. When I write WITs with random experiments, I just end up checking the JSON structure is okay, rather than the content. I do think we should check for content, like you do here. But perhaps eventually we would have files with expected JSON responses for each experiment in the fixtures? I'm not sure what the best way of doing it would be.

Reply:
This is from a time when we were writing worse tests. Some of them I parameterized, others I just changed the hard-coded actual/expected values. In principle yes, the best thing would be to test structure or that the values match a set of possible values.